src/NHibernate.Test/Tools/hbm2ddl/SchemaMetadataUpdaterTest/SchemaMetadataUpdaterFixture.cs

```diff
@@ -56,10 +59,98 @@ public void UpdateReservedWordsInDialect()
 			var sf = (ISessionFactoryImplementor) configuration.BuildSessionFactory();
 			SchemaMetadataUpdater.Update(sf);
 			var match = reservedDb.Intersect(sf.Dialect.Keywords);
+
+			// tests that nothing in the first metaData.GetReservedWords() is left out of the second metaData.GetReservedWords() call.
+			// i.e. always passes.
+			Assert.That(match, Is.EquivalentTo(reservedDb));
+		}
+
+		[Test]
+		public void EnsureReservedWordsHardCodedInDialect()
+		{
+			var reservedDb = new HashSet<string>();
+			var configuration = TestConfigurationHelper.GetDefaultConfiguration();
+			var dialect = Dialect.Dialect.GetDialect(configuration.Properties);
+			var connectionHelper = new ManagedProviderConnectionHelper(configuration.Properties);
+			connectionHelper.Prepare();
+			try
+			{
+				var metaData = dialect.GetDataBaseSchema(connectionHelper.Connection);
+				foreach (var rw in metaData.GetReservedWords())
+				{
+					if (rw.Contains(" ")) continue;
+					reservedDb.Add(rw.ToLowerInvariant());
+				}
+			}
+			finally
+			{
+				connectionHelper.Release();
+			}
+
+			var sf = (ISessionFactoryImplementor)configuration.BuildSessionFactory();
+
+			// use the dialect as configured, with no update
+			var match = reservedDb.Intersect(sf.Dialect.Keywords).ToList();
+
+			// tests that nothing in metaData.GetReservedWords() is left out of the Dialect.Keywords (without a refresh).
+			var differences = reservedDb.Except(match).ToList();
+			if (differences.Count > 0)
+			{
+				Console.WriteLine("Update Dialect {0} with RegisterKeyword:", sf.Dialect.GetType().Name);
+				foreach (var keyword in differences.OrderBy(x => x))
+				{
+					Console.WriteLine("  RegisterKeyword(\"{0}\");", keyword);
+				}
+			}
+
+			if (sf.ConnectionProvider.Driver is OdbcDriver)
+			{
+				Assert.Inconclusive("ODBC has excess keywords reserved");
```

Comment:
This one a merge, you can rebase/merge master for avoiding `Assert.Inconclusive` causing a failure for builds using results comparison.